### PR TITLE
[GH-748] Upgraded IDNA library and adjusted NameUtil module

### LINKED ISSUE #748: Upgrade IDNA library
IDNA library should be upgraded to 6.0.0. Adjustments should be made to `normalize_and_validate_name` function in Naming Utils module.[link](https://github.com/aeternity/epoch/commit/3803f9e8a344a321469822405757ab3650ee4f6a#diff-6da3f427a39bebc64c43177ec6351ffaR75)

### PR DESCRIPTION
**Pull request by DanielaIvanova**
resolves #748

### CODE REVIEW

**Review by thepiwo — CHANGES_REQUESTED** [review]

File: apps/aecore/lib/aecore/naming/name_util.ex [hunk 1]
```diff
@@ -19,16 +19,36 @@ defmodule Aecore.Naming.NameUtil do
 
   @spec normalize_and_validate_name(String.t()) :: {:ok, String.t()} | {:error, reason()}
   def normalize_and_validate_name(name) do
-    normalized_name = normalize_name(name)
+    case validate_normalized_name(name) do
+      :ok ->
+        {:ok, normalize_name(name)}
```
[the remark this answers is not in the document]
Comment (DanielaIvanova) [reply apps/aecore/lib/aecore/naming/name_util.ex]: `Epoch` does the same. You can [check](https://github.com/aeternity/epoch/blob/v0.25.0/apps/aens/src/aens_utils.erl#L38)

File: apps/aecore/lib/aecore/naming/name_util.ex [hunk 2]
```diff
@@ -19,16 +19,36 @@ defmodule Aecore.Naming.NameUtil do
...
+
+    name_split_count =
+      name_unicode_list
+      |> :idna.encode([{:uts46, true}, {:std3_rules, true}])
+      |> to_string()
+      |> String.split(GovernanceConstants.split_name_symbol())
+      |> length()
+
+    if name_split_count == GovernanceConstants.name_split_check() do
+      List.to_string(name_unicode_list)
```
[the remark this answers is not in the document]
Comment (DanielaIvanova) [reply apps/aecore/lib/aecore/naming/name_util.ex]: You are right! I will fix it.

**Review by thepiwo — APPROVED** [review]
seems good

File: apps/aecore/lib/aecore/naming/name_util.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (d-velev) [reply apps/aecore/lib/aecore/naming/name_util.ex]: Could you rename the function to validate_name instead so it's not confusing?

**Review by Artur64 — APPROVED** [review]

**Review by cytadela8 — APPROVED** [review]

**Pull request merged**